### DISCUSSION TITLE
Compare .xml files automatically as XML

### DISCUSSION THREAD

**Discussion by igitur** [WinMerge/winmerge#657] [Ideas]
Hi,

When comparing .csv files, WinMerge automatically compares them as tables, which makes sense. Similarly, when comparing .xml files, I think they should automatically be compared as XML instead of TEXT. If would save me quite a few keystrokes as I'm busy reverse-engineering many OpenXml XML files (which go inside Excel's .xlsx files).

thanks

**Comment by mlipok** [comment]
Are you asking about XML syntax highlighting?

**Reply by igitur** [reply]
No, rather the pretty-printing that one gets when selecting:
![image](https://user-images.githubusercontent.com/145854/109622430-c51df700-7b44-11eb-830e-dbfd66e52284.png)

As I say, for .csv files, they are automatically compared as tables (from that same submenu).

**Comment by JhonnyPala** [comment]
Me too! I would be happy if you implemented this feature, both from the settings and as a command line parameter

**Reply by oelgoetz** [reply]
Yes. Having it as a command line parameter would be extremely useful!

**Comment by sdottaka** [comment]
Copy the following DisplayXMLFiles.dll to C:\Program Files\WinMerge\MergePlugins folder and select the Plugins → Automatic Unpacking menu item and it should work as you would expect.

https://github.com/WinMerge/winmerge/blob/master/Plugins/dlls/DisplayXMLFiles.dll

![image](https://user-images.githubusercontent.com/98126/110208138-b6c83600-7eca-11eb-846e-ddf3f20d4f3a.png)

**Reply by igitur** [reply]
Thanks. I remember that plugin. I thought it was obsolete. So I'll try it, but it does feel like a workaround. Can you not handle automatic XML unpacking like you do with CSV files?

**Comment by JhonnyPala** [comment]
Thank you!
Better than nothing.
Cheers

Il giorno sab 6 mar 2021 alle ore 14:27 Takashi Sawanaka <
***@***.***> ha scritto:
